jit/src/main/java/org/dcsa/conformance/standards/jit/action/JitAction.java

```diff
@@ -1,30 +1,51 @@
 package org.dcsa.conformance.standards.jit.action;
 
-import java.util.Map;
-import java.util.function.Supplier;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
 import org.dcsa.conformance.core.scenario.ConformanceAction;
-import org.dcsa.conformance.standards.jit.party.SuppliedScenarioParameters;
+import org.dcsa.conformance.standards.jit.party.DynamicScenarioParameters;
 
+@Slf4j
 public abstract class JitAction extends ConformanceAction {
-  protected final Supplier<SuppliedScenarioParameters> sspSupplier;
-  protected final int expectedStatus;
 
-  public JitAction(
+  @Getter protected DynamicScenarioParameters dsp;
+
+  protected JitAction(
       String sourcePartyName,
       String targetPartyName,
       ConformanceAction previousAction,
-      String actionTitle,
-      int expectedStatus) {
+      String actionTitle) {
     super(sourcePartyName, targetPartyName, previousAction, actionTitle);
-    this.sspSupplier = _getSspSupplier(previousAction);
-    this.expectedStatus = expectedStatus;
+    if (previousAction instanceof JitAction previousJitAction && previousJitAction.dsp != null) {
+      dsp = previousJitAction.dsp;
+    }
+  }
+
+  @Override
+  public void reset() {
+    super.reset();
+    if (previousAction != null) {
+      dsp = null;
+    }
+  }
```

Comment:
`reset` does not seem consistent with the constructor on how `dsp` is being initialized. How does this work?

Reply:
You're right. I'll try to improve it as well.